### ISSUE TITLE
Messages appear duplicated

### ISSUE THREAD

**Issue by Skullmonkey** [klinker-apps/messenger-issues#517]
**Device with issue:** Pixel XL Android 8.1
**Primary device:** Pixel XL Android 8.1
**App version:** 2.63.1743
**Steps to reproduce:**

1. Unknown

**Additional information:**

Sometimes, it appears that messages are duplicated even though they actually say different things. I am not able to accurately duplicate the problem. I find that it seems to happen most when I send and receive a message at the same time. It always happens when there are multiple sent or received messages in a row. For example earlier, I opened a conversation and I had 2 messages received. One was "Mmk" and the other was "When lunch is over". Both messages displayed as "Mmk" until I closed the conversation and reopened it. It then displayed as it should. I have also noticed that scrolling up in the conversation and scrolling back fixes it.

**Comment by rattrick1** [comment]
I see this message every single day. I have the stock Google Messenger app installed and when I open the same conversation in there, it shows me the correct (missing) messages. This to me says the OS SMS stuff is working fine and it's likely an issue with Pulse syncing the message that's causing it to show duplicated.

**Comment by klinker24** [comment]
I don't think you are talking about the same issue as described here. This issue goes away by just scrolling or closing and reopening the thread.

**Comment by rattrick1** [comment]
Yes, you're right. Mine are wrong/missing forever, no matter what.

**Comment by enonymous1** [comment]
I'd like to add to this. It sounds very similar if not the exact issue I'm having. I'll also disclose I'm running this on a Verizon Google pixel 2 xl. Often times as stated above when receiving multiple messages the first message notification is great. It shows the sender and message details. If a second message comes in before I've cleared that notification or views the message the 2nd message will appear to be a duplicate of the message that's been unread. When in fact if I open the app I have two separate messages. It doesn't seem to matter if the messages are from the same person or not, it's just if one comes in and goes unread then the 2nd is just a duplicate. I'm thankful that hey I get a notification and I know I got another message but can be very misleading as to the actual message.

Along the same path if person 1 texts me and i don't actually open it and a 2nd messages me and I use the reply feature from the notification bar (not actually open pulse and then message) the reply will go to the first person and not the 2nd. I'm glad I caught this early as I could've easily gotten myself in to some bad situations, I simply don't use the reply feature from the notification bar because of this issue though.

**Comment by Skullmonkey** [comment]
I believe that you are discussing a different issue. My issue happens and is continuing to happen while I have a conversation open. It really has nothing to do with any notifications. The only thing I can really pin on  it happening is that the other person that you are talking to sends more than one message in a row, I notice it more when it is in rapid succession but I have also noticed it while having the conversation open a while. Its really hard to explain to get a clear picture of whats going on.

Basically when you have a conversation open, it goes like this:
> Sender: This is message 1.

and then they send another text saying "This is message 2", the conversation will update and say:
> Sender: This is message 1.
> Sender: This is message 1.

But when you scroll past the message or close the conversation, it will go to what it should say like: 
> Sender: This is message 1.
> Sender: This is message 2.

**Comment by core1211** [comment]
I'm having the issue where if I get a text and then another text, the first text will show up every time I get a new text until I view it. This is a never ending loop for every time I get a new text, any that follow from either the same person or anyone else will continue to only show the very first text I received. Very annoying.

**Comment by Skullmonkey** [comment]
I believe you are experiencing a different thing than I am. I think I know what you are talking about and that is a setting thing. I would open up a new issue for help on that one.

**Comment by klinker24** [comment]
Please don't open issues for questions about settings. If you have a question about the app, email me instead, or even better, check out my help page. It is probably answered there. I don't know what you are asking about, based on the comment, sorry.

https://messenger.klinkerapps.com/help

**Comment by Skullmonkey** [comment]
I believe he is talking about notifications. I know when I receive texts from multiple people, I will always receive a new notification when I have one conversation open but have not read the other (if that makes sense) until I either open the conversation or mark it as read via the notification action. But that would be intended usage in my eyes. 

So:
Person 1 send me a text. I leave it alone.
Person 2 sends me a text that I open. 
Person 2 sends another text and I receive a notification acting like person 1 has sent me another but in actuality has not. This happens until you mark person 1's conversation as read or you open that conversation.

If this is the 'problem' he (Core) is having, I remember you (Luke) saying at some point that it is an issue on how Android handles notifications and there is nothing that you can do about it.

**Comment by klinker24** [comment]
@core1211 #636

**Comment by nuckjr40** [comment]
I'm seeing duplicate messages from the same person in addition to seeing read messages in the notification window. For example, I receive a new message and the notification will show two or three previously read messages.

**Comment by klinker24** [comment]
There is an option in settings to turn off the history in notifications. That isn't what this is issue is talking about though.

https://messenger.klinkerapps.com/help

**Comment by nuckjr40** [comment]
History is one thing but that doesn't answer the question why I'm seeing duplicate messages in the app.

**Comment by klinker24** [comment]
That's why the issue is still open.

**Comment by dmcrider** [comment]
Not sure if my issue is the same, but I think it's similar:

When I schedule a message, it always shows up as sent twice with the same timestamp. Shows up in the App and on the web. Haven't verified yet if the receiver gets two messages or not.

Let me know if this is too different so I can open an issue!

**Comment by egocarib** [comment]
I believe that this issue might be related to when a conversation reaches it's maximum message capacity. I have a large group conversation and it only holds about 3 months of messages, everything before that is cut off (I assume because Pulse may enforce a maximum message # for conversations).

I started hitting this issue when my conversation hit that limit. Now whenever I get new messages and open the conversation, the recent messages are duplicated or appear to overwrite one another. The issue resolves itself if I scroll up and then back down - then I see the correct messages at the end of the queue.

**Comment by rovindrasingh** [comment]
Has any headway been made on this? I'm starting to notice this issue in group messages now myself. Whenever someone sends a message every single message in conversation view gets duplicated and disappears upon either scrolling or closing and re-opening the conversation. This is getting pretty frustrating.

**Comment by rattrick1** [comment]
I made headway, I switched to Android Messages and haven't missed a message since. The dev doesn't care to investigate, but I'm positive it's caused by messages that couldn't be received due to bad service.

**Comment by klinker24** [comment]
It seems like I have only ever hear about this issue with group messages, and individual conversations work fine. Not sure if that is accurate or not, but I am unable to reproduce any issues with individual conversation (or group conversations, admittedly). Yes, I am sure that the issue is with the size limit of what is displayed in conversations.

I have made a change for the next version of the app. I cannot confirm whether it fixes anything, since I do not see the issue myself.

**Comment by rovindrasingh** [comment]
Hey Luke, I can confirm that it doesn't act that way for individual conversations, only for group conversations. I just received the latest update (4.7.0.2399) that includes the new color scheme and unfortunately the issue is still present for me. Is there anything I could possibly do on my end to attempt to rectify this? I'm running this on my Pixel 3 XL, pure stock.

**Comment by klinker24** [comment]
You could go to the settings menu for that conversation and use the option to delete old messages.

**Comment by egocarib** [comment]
Is there more information we can provide to help you investigate this issue?
For me at least, group conversations are the main use case I have for this app. If this issue cannot be resolved I will likely need to look for another messaging app. Would prefer not to do that because I like everything else about Pulse.

**Comment by klinker24** [comment]
Deleting messages is not a great workaround, no. It is the only thing I have right now though, sorry. 

This won't happen until the conversation reaches more than 8,000 messages. I do not think that many people would be referring to messages that are back that far.

**Comment by Skullmonkey** [comment]
Mine is doing it for an individual conversation. It is quite lengthy since it has definitely been cutting off messages past a certain point (only going to Sep 16, 2018) and I have been using this app way longer than that without resetting it.

**Comment by klinker24** [comment]
As noted in my previous comment, your issue is different from the one that we are currently discussing. I have fixed your issue, for the next version.

**Comment by Skullmonkey** [comment]
Ah sorry. Been sick the past few days so I am not thinking clearly.

**Comment by klinker24** [comment]
I have also made changes around the conversation limit, in group conversations, for the next version of the app. It should fix this issue. If not, someone can comment back and I will re-open it.

**Comment by egocarib** [comment]
The update definitely changed the behavior somewhat, but the issue still occurs. Messages disappear temporarily or are duplicated until I scroll up/down to restore the view.

Here's an image of the problem as it unfolded mid conversation:
![2019-03-19_13-51-41](https://user-images.githubusercontent.com/3429497/54637974-eec4dd00-4a57-11e9-90dc-c38cb13c2755.png)

**Comment by klinker24** [comment]
I will open a new issue for this, when I get some time. Thanks for the report and the screenshots
